airflow/providers/apache/spark/hooks/spark_submit.py

```diff
@@ -237,8 +237,8 @@ def _mask_cmd(self, connection_cmd):
         # Mask any password related fields in application args with key value pair
         # where key contains password (case insensitive), e.g. HivePassword='abc'
         connection_cmd_masked = re.sub(
-            r"(\S*?(?:secret|password)\S*?\s*=\s*')[^']*(?=')",
-            r'\1******', ' '.join(connection_cmd), flags=re.I)
+            r"(\S*?(?:secret|password)\S*?\s*(?:=|\s+)(['\"]?))[^'^\"]+(\2)",
```

Comment:
Does this regular expression deserve some more explaining, possibly in the form of [inline comments](https://docs.python.org/3/library/re.html#re.VERBOSE)?

Reply:
It would be nice 🙂

Reply:
Done. Let me know which parts aren't clear enough. :)